library/Icingadb/Widget/ItemList/RedundancyGroupListItem.php

```diff
@@ -0,0 +1,176 @@
+            ->first();
+
+        $this->isHandled = $this->state->failed
+            && (
+                $this->summary->objects_problem_handled
+                || $this->summary->objects_unknown_handled
+                || $this->summary->objects_warning_handled
+            );
+    }
+
+    protected function getStateBallSize(): string
+    {
+        return StateBall::SIZE_LARGE;
+    }
+
+    protected function createTimestamp(): ?BaseHtmlElement
+    {
+        return new TimeSince($this->state->last_state_change->getTimestamp());
+    }
+
+    protected function createSubject(): BaseHtmlElement
+    {
+        return new HtmlElement(
+            'span',
+            Attributes::create(['class' => 'subject']),
+            Text::create($this->item->display_name)
+        );
+    }
+
+    protected function assembleVisual(BaseHtmlElement $visual): void
+    {
+        $stateBall = new StateBall($this->state->getStateText(), $this->getStateBallSize());
+        if ($this->isHandled) {
+            $stateBall->getAttributes()->add('class', 'handled');
+        }
+
+        $visual->addHtml($stateBall);
+    }
+
+    protected function assembleCaption(BaseHtmlElement $caption): void
+    {
+        $members = RedundancyGroup::on($this->getDb())
+            ->columns([
+                'id' => 'id',
+                'objects_output' => new Expression(
+                    'CASE WHEN redundancy_group_from_to_host_state.output IS NULL'
+                    . ' THEN redundancy_group_from_to_service_state.output'
+                    . ' ELSE redundancy_group_from_to_host_state.output END'
+                ),
+                'objects_long_output' => new Expression(
+                    'CASE WHEN redundancy_group_from_to_host_state.long_output IS NULL'
+                    . ' THEN redundancy_group_from_to_service_state.long_output'
+                    . ' ELSE redundancy_group_from_to_host_state.long_output END'
+                ),
+                'objects_checkcommand_name' => new Expression(
+                    'CASE WHEN redundancy_group_from_to_host.checkcommand_name IS NULL'
+                    . ' THEN redundancy_group_from_to_service.checkcommand_name'
+                    . ' ELSE redundancy_group_from_to_host.checkcommand_name END'
+                ),
+                'objects_last_state_change' => new Expression(
+                    'CASE WHEN redundancy_group_from_to_host_state.last_state_change IS NULL'
+                    . ' THEN redundancy_group_from_to_service_state.last_state_change'
+                    . ' ELSE redundancy_group_from_to_host_state.last_state_change END'
+                ),
+                'objects_severity' => new Expression(
+                    'CASE WHEN redundancy_group_from_to_host_state.severity IS NULL'
+                    . ' THEN redundancy_group_from_to_service_state.severity'
+                    . ' ELSE redundancy_group_from_to_host_state.severity END'
+                )
```

Comment:
Don't you fetch only host state details here? `redundancy_group_from_to_host_state` should always succeed for services.